solr/core/src/java/org/apache/solr/update/processor/ParseDateFieldUpdateProcessorFactory.java

```diff
@@ -120,6 +121,16 @@ public UpdateRequestProcessor getInstance(SolrQueryRequest req,
       protected Object mutateValue(Object srcVal) {
         if (srcVal instanceof CharSequence) {
           String srcStringVal = srcVal.toString();
+          // trim single quotes around date if present
```

Comment:
Oh yeah.  It's unfortunate such ugly data comes in as such... but it's at least cheap & safe to do this so I guess it's okay.